lib/alephant/sequencer/sequence_cache.rb

```diff
@@ -0,0 +1,69 @@
+require "dalli-elasticache"
+require "alephant/logger"
+
+module Alephant
+  module Sequencer
+    class SequenceCache
+      include Logger
+
+      attr_reader :config
+
+      DEFAULT_TTL  = 5
+
+      def initialize(config={})
+        @config = config
+
+        unless config_endpoint.nil?
+          @elasticache ||= ::Dalli::ElastiCache.new(config_endpoint, { :expires_in => ttl })
+          @client ||= @elasticache.client
+        else
+          logger.debug "Alephant::SequenceCache::#initialize: No config endpoint, NullClient used"
+          logger.metric "NoConfigEndpoint"
+          @client = NullClient.new
+        end
+      end
+
+      def get(key, &block)
+        begin
+          versioned_key = versioned key
+          result = @client.get versioned_key
+          logger.info "Alephant::SequenceCache#get key: #{versioned_key} - #{result ? 'hit' : 'miss'}"
+          logger.metric "GetKeyMiss" unless result
+          result ? result : set(key, block.call)
+        rescue StandardError => e
+          block.call if block_given?
```

Comment:
I did see this the other day, and I assumed that the error would be caught by the rescue if a block wasn't passed in. At the moment, the get is only called within the Sequencer/Sequencer class and a block if provided.

Is it worth removing the `block_given?` check, as the block is provided, and I can't see a case where we would add functionality and not provide a block if the item wasn't in the cache.